### PR TITLE
feat: add yaml language extension

### PR DESCRIPTION
**Pull request by appelgriebsch**
add `yaml` language extension incl. yaml schema companion

### CODE REVIEW

File: lua/lazyvim/plugins/extras/lang/yaml.lua [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      "someone-stole-my-name/yaml-companion.nvim",
+      version = false, -- last release is way too old
+      config = function(_, opts)
+        require("telescope").load_extension("yaml_schema")
+      end,
+    },
+    opts = {
+      -- make sure mason installs the server
+      servers = {
+        yamlls = {}
```
Comment (mochaaP) [inline lua/lazyvim/plugins/extras/lang/yaml.lua:26]: Should use `SchemaStore` here:

```lua
      yamlls = {
        ---@param new_config lspconfig.options.yamlls
        on_new_config = function(new_config)
          new_config.settings.yaml.schemas = new_config.settings.yaml.schemas or {}
          vim.list_extend(new_config.settings.yaml.schemas, require("schemastore").yaml.schemas())
        end,
      },
```

File: lua/lazyvim/plugins/extras/lang/yaml.lua [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        vim.list_extend(opts.ensure_installed, { "yaml" })
+      end
+    end,
+  },
+
+  -- correctly setup lspconfig
+  {
+    "neovim/nvim-lspconfig",
+    dependencies = {
+      "someone-stole-my-name/yaml-companion.nvim",
```
Comment (mochaaP) [inline lua/lazyvim/plugins/extras/lang/yaml.lua:17]: Is this really required for those who don't need K8s integration?

File: lua/lazyvim/plugins/extras/lang/yaml.lua (hunk 1 above)
Comment (mochaaP) [reply lua/lazyvim/plugins/extras/lang/yaml.lua:26]: also turn telemetry off by default:
```lua
{ telemetry = { enabled = false } },
```

**Review by mochaaP — CHANGES_REQUESTED** [review]

File: lua/lazyvim/plugins/extras/lang/yaml.lua (hunk 1 above)
Comment (appelgriebsch) [reply lua/lazyvim/plugins/extras/lang/yaml.lua:26]: its the default for the `yaml-companion...`: https://github.com/someone-stole-my-name/yaml-companion.nvim#%EF%B8%8F--configuration

Comment (mochaaP) [reply lua/lazyvim/plugins/extras/lang/yaml.lua:26]: `yaml-companion`'s builtin SchemaStore integration does not provides local cache on the catalog and may possibly cause slowdown on file matching. `SchemaStore.nvim` does auto-generate with CI: <https://github.com/b0o/SchemaStore.nvim/blob/main/lua/schemastore/catalog.lua>

Also this brings better consistency with <https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/plugins/extras/lang/json.lua>.

Comment (appelgriebsch) [reply lua/lazyvim/plugins/extras/lang/yaml.lua]: Thanks, updated according to README at https://github.com/b0o/SchemaStore.nvim

File: lua/lazyvim/plugins/extras/lang/yaml.lua (hunk 2 above)
Comment (appelgriebsch) [reply lua/lazyvim/plugins/extras/lang/yaml.lua]: removed

**Pull request merged**